src/Abuse/Adapters/TimeLimit.php

```diff
@@ -234,8 +234,7 @@ protected function hit(string $key, string $datetime): void
                 ];
                 $this->db->createDocument(TimeLimit::COLLECTION, new Document($data));
             } else {
-                $data->setAttribute('count', $data->getAttribute('count', 0) + 1);
-                $this->db->updateDocument(TimeLimit::COLLECTION, $data->getId(), $data);
+                $this->db->increaseDocumentAttribute(TimeLimit::COLLECTION, $data->getId(),'count');
```

Comment:
This solves the problem where multiple hits are called at the same time and they all get incremented to the same value.

However, this doesn't solve the problem where no document exists yet so multiple hits try to call `$this->db->createDocument(TimeLimit::COLLECTION, new Document($data));`

Reply:
We have a unique Index on key + time .
So on race conditions potentionially one hit can be a unique duplicate error.

Reply:
@eldadfux @stnguyen90 @TorstenDittmann 
Do you think the try - catch will solve the race condition?

Reply:
@fogelito I do think this will help with the duplicate error. What's the plan for the Appwrite side?

https://github.com/appwrite/appwrite/blob/de8552e9018df460d07bdbd587716b95e0c9f083/app/controllers/shared/api.php#L148-L167

Reply:
@stnguyen90 Can you clarify the remaining issue from the Appwrite side?

Reply:
IIRC, the problem on the Appwrite side is that if multiple API calls are made simultaneously, they can all incorrectly return the same `X-RateLimit-Remaining` header.

Reply:
I guess that's 2 requests running simultaneously with the count method on the exact millisecond?
If this is the case, it is possible to control the reads by changing Isolation level on the transaction or to add select for update , which can lock the row from reading while updating to control race conditions.
@eldadfux I remember saying something about this issue?

Reply:
If we have a duplicated error on the unique index, we should update the document using the increase method.

Reply:
This is the exact fix for the race condition, but I am referring what steven mentioned on the Appwrite side about the 
X-RateLimit-Remaining Header race condition?

Reply:
@fogelito, I think to really solve this, the `increaseDocumentAttribute()` method **must** return the updated count. That way, the count won't be stale.

Reply:
> @fogelito, I think to really solve this, the `increaseDocumentAttribute()` method **must** return the updated count. That way, the count won't be stale.

@eldadfux What do you think about returning the updated value from the increase?

Reply:
See https://github.com/utopia-php/abuse/pull/37#issuecomment-1459930027